tests/momento/simple_cache_client/test_scalar_async.py

```diff
@@ -9,274 +12,268 @@
-    assert isinstance(set_resp, CacheSet.Success)
-
-    get_resp = await client_async.get(cache_name, key)
-    assert isinstance(get_resp, CacheGet.Hit)
-    assert get_resp.value_bytes == value
-
-
-async def test_get_returns_miss(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key = uuid_str()
-
-    get_resp = await client_async.get(cache_name, key)
-    assert isinstance(get_resp, CacheGet.Miss)
-
-
-async def test_expires_items_after_ttl(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key = uuid_str()
-    val = uuid_str()
-
-    await client_async.set(cache_name, key, val, timedelta(seconds=2))
-    get_response = await client_async.get(cache_name, key)
-    assert isinstance(get_response, CacheGet.Hit)
-
-    time.sleep(4)
-    get_response = await client_async.get(cache_name, key)
-    assert isinstance(get_response, CacheGet.Miss)
-
-
-async def test_set_with_different_ttl(client_async: SimpleCacheClientAsync, cache_name: str) -> None:
-    key1 = uuid_str()
-    key2 = uuid_str()
-
-    await client_async.set(cache_name, key1, "1", timedelta(seconds=2))
-    await client_async.set(cache_name, key2, "2")
-
-    # Before
-    get_response = await client_async.get(cache_name, key1)
-    assert isinstance(get_response, CacheGet.Hit)
-    get_response = await client_async.get(cache_name, key2)
-    assert isinstance(get_response, CacheGet.Hit)
+def a_cache_name_validator():
+    async def with_non_existent_cache_name_it_throws_not_found(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        cache_name = uuid_str()
+        response = await cache_name_validator(cache_name)
+        assert response.error_code == MomentoErrorCode.NOT_FOUND_ERROR
+
+    async def with_null_cache_name_it_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        response = await cache_name_validator(None)
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache name must be a non-empty string"
+
+    async def with_empty_cache_name_it_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name_validator
+    ) -> None:
+        response = await cache_name_validator("")
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache header is empty"
+
+    async def with_bad_cache_name_throws_exception(client_async: SimpleCacheClientAsync, cache_name_validator) -> None:
+        response = await cache_name_validator(1)
+        assert response.error_code == MomentoErrorCode.INVALID_ARGUMENT_ERROR
+        assert response.inner_exception.message == "Cache name must be a non-empty string"
+
+
+def a_key_validator():
+    async def with_null_key_throws_exception(
+        client_async: SimpleCacheClientAsync, cache_name: str, key_validator
```

Comment:
We would also need annotations for the various validators. That may prove tricky.

Reply:
This is where the lack of a common response class hierarchy bites us. I think we'll be good enough by changing `class CacheGetResponse(ABC):` to `class CacheGetResponse(Response):` and `class Response(ABC)`. Then these would return Response.